worker/uniter/runner/runner.go

```diff
@@ -214,19 +246,16 @@ func (runner *runner) RunAction(actionName string) error {
 	if actionName == actions.JujuRunActionName {
 		return runner.runJujuRunAction()
 	}
-	return runner.runCharmHookWithLocation(actionName, "actions")
+	// run actions on remote workload pod for caas.
+	return runner.runCharmHookWithLocation(actionName, "actions", runner.context.ModelType() == model.CAAS)
 }
 
 // RunHook exists to satisfy the Runner interface.
 func (runner *runner) RunHook(hookName string) error {
-	return runner.runCharmHookWithLocation(hookName, "hooks")
+	return runner.runCharmHookWithLocation(hookName, "hooks", false)
 }
 
-func (runner *runner) runCharmHookWithLocation(hookName, charmLocation string) error {
-	if runner.context.ModelType() == model.CAAS {
-		// TODO(caas): run on workload.
-	}
-
+func (runner *runner) runCharmHookWithLocation(hookName, charmLocation string, runOnRemote bool) (err error) {
```

Comment:
Instead of a bool for RunOnRemote, it's often clearer to define typed consts for local/remote and pass those in. We do that elsewhere for model config (WithDefauts) etc

Reply:
fixed